fabfile/alba.py

```diff
@@ -79,29 +84,122 @@ def arakoon_start_(arakoon_config_file, base_dir, arakoon_nodes):
+    root = TLS['root_dir']
+    where = local
+    where ("mkdir -p %s" % root)
+    with lcd(root):
+        subject = \
+            '/C=BE/ST=Vl-Br/L=Leuven/O=openvstorage.com/OU=AlbaTest/CN=AlbaTest CA'
+        where("openssl req -new -nodes -out cacert-req.pem -keyout cacert.key -subj '%s'" % subject)
+
+        # Self-sign CA CSR
+        where ('openssl x509 -signkey cacert.key -req -in cacert-req.pem -out cacert.pem')
+        where ("rm cacert-req.pem")
+
+
+@task
+def make_cert(name, serial = None):
+    # CSR
+    root = TLS['root_dir']
+    node_path = '%s/%s' % (root, name)
+    where = local
+    where("mkdir -p %s" % node_path)
+    with lcd(node_path):
+        subject = \
+            '/C=BE/ST=Vl-BR/L=Leuven/O=openvstorage.com/OU=AlbaTest/CN=%s' % name
+        cmd = "openssl req -out %s-req.pem -new -nodes -keyout %s.key" % (name,name)
+        cmd += " -subj '%s'" % subject
+        where(cmd)
+
+        # Sign
+        cmd = "openssl x509 -req -in %s-req.pem -CA %s/cacert.pem -CAkey %s/cacert.key -out %s.pem" \
+              % (name, root, root, name)
+
+        if serial:
+            cmd += " -set_serial 0%(serial)d" % {'serial': serial}
+
+        # persist serial number state
+        cmd += " -CAcreateserial -CAserial %s/cacert-serial.seq" % root
+
+
+        where(cmd)
+
+        where("rm %s-req.pem" % name)
+
+        # Verify
+        where('openssl verify -CAfile %(root)s/cacert.pem %(name)s.pem' % \
+              {'name': name, 'root':root }
+        )
 
 @task
-def arakoon_who_master(arakoon_cfg_file = arakoon_config_file):
+def arakoon_start(tls, env = env):
+    acf = arakoon_config_file
+    where = local
+    path = "/tmp/arakoon"
+    where ("mkdir -p %s" % path)
+    if _use_tls(tls):
+        acf = TLS['arakoon_config_file']
+        make_ca()
+        serial = 0
+        for name in arakoon_nodes:
+            make_cert(name = name, serial = serial)
+            serial = serial + 1
+
+        with lcd(path):
+            client = "my_client"
+            where("mkdir ./%s" % client)
+            make_cert(name = client, serial = serial)
+
+    arakoon_start_(acf, "/tmp/arakoon", arakoon_nodes, env = env)
+
+def _extend_arakoon_tls(cmd):
+    root = TLS['root_dir']
+    name = "my_client"
+    cmd.extend([
+        '-tls-ca-cert', "%s/cacert.pem" % root,
+        '-tls-cert',    "%s/%s/%s.pem"     % (root, name, name),
+        '-tls-key',     "%s/%s/%s.key"     % (root, name, name)
+    ]
+    )
+
+def _extend_alba_tls(cmd):
+    root = TLS['root_dir']
+    name = "my_client"
+    cmd.extend([
+        '--tls=%s,%s,%s' % ('%s/cacert.pem' % root,
+                            '%s/%s/%s.pem' % (root, name, name),
+                            '%s/%s/%s.key' % (root, name, name))
+    ])
+@task
+def arakoon_who_master(arakoon_cfg_file = arakoon_config_file,
+                       tls = 'False',
+                       env = env):
+    if _use_tls(tls):
+        arakoon_cfg_file = TLS['arakoon_config_file']
```

Comment:
```
def arakoon_who_master(arakoon_cfg_file = None,
                                            ....
    if arakoon_cfg_file is None:
      arakoon_cfg_file = _use_tls(tls) ? TLS['arakoon_config_file'] : arakoon_config_file
```
(ok, I admit that's not valid python)
the point is that using tls doesn't mean the method should ignore the specified config file